pkg/psmdb/backup/pbm.go

```diff
@@ -334,6 +335,46 @@ func GetPBMConfig(ctx context.Context, k8sclient client.Client, cluster *api.Per
 	return conf, nil
 }
 
+func GetPBMStorageMinioConfig(
+	ctx context.Context,
+	k8sclient client.Client,
+	cluster *api.PerconaServerMongoDB,
+	stg api.BackupStorageSpec,
+) (config.StorageConf, error) {
+	storageConf := config.StorageConf{
+		Type: storage.Minio,
+		Minio: &mio.Config{
+			Region:                stg.Minio.Region,
+			Endpoint:              stg.Minio.EndpointURL,
+			Bucket:                stg.Minio.Bucket,
+			Prefix:                stg.Minio.Prefix,
+			InsecureSkipTLSVerify: stg.Minio.InsecureSkipTLSVerify,
+			DebugTrace:            stg.Minio.DebugTrace,
+			PartSize:              stg.Minio.PartSize,
+			Secure:                stg.Minio.Secure,
+		},
+	}
+
+	if len(stg.Minio.CredentialsSecret) != 0 {
+		s3secret, err := getSecret(ctx, k8sclient, cluster.GetNamespace(), stg.Minio.CredentialsSecret)
+		if err != nil {
+			return storageConf, errors.Wrap(err, "get minio credentials secret")
```

Comment:
why we return a partially initialized storage conf instead of empty struct? i.e. `config.StorageConf{}`

Reply:
updated